core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -455,22 +455,9 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
         producer.flush()
         assertTrue("All request are complete.", responses.forall(_.isDone()))
         // Check the messages received by broker.
-        val records = new ArrayList[ConsumerRecord[Array[Byte], Array[Byte]]]()
-        val maxIters = 50
-        var iters = 0
-        consumer1.seek(new TopicPartition(topic, 0),0)
```

Comment:
To close the loop on this one, using `seek` makes the test a lot slower, so the approach in the PR where we just consume the records that have been produced since the last iteration is better.